envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -88,18 +104,16 @@ class EnvoyClustersFactory(
     }
 
     private fun selectCluster(group: Group, globalSnapshot: GlobalSnapshot, clusterName: String): Cluster? {
```

Comment:
It think `selectCluster` and `enableTlsForCluster` are no longer describing the reality.
* `enableTlsForCluster` - doesn't make sense, we don't enable tls per cluster, but per group (cluster is not even a parameter of this method anymore)
* `selectCluster` - we no longer need to select cluster by cluster. We can just check, if mtls is enabled for given group and then we can return clusters from `globalSnapshot.clusters` or `globalSnapshot.securedClusters`